### ISSUE TITLE
Unable to load regular npm modules from withing component

### ISSUE THREAD

**Issue by DerJacques** [nblackburn/vue-brunch#6]
Hi there

First: Thank you so much for continuously working on Vue Brunch. We're really happy to use it.

After upgrading to Vue 2.0 and Vue Brunch 1.2.1, I can no longer import any regular npm modules from within my components.

Here's the error:
`Uncaught Error: Cannot find module 'lodash' from 'web/static/js/components/task.vue'`

The relevant section from `task.vue`:
```
<script>
  import state from '../state'
  import _ from 'lodash'

  module.exports = {
```

Note that I can run `import _ from 'lodash'` from my `main.js` (so outside Vue) just fine.
Even stranger to me: If I add `import _ from 'lodash'` to my `main.js`, the same `import _ from 'lodash'` now also works in my Vue components.

I'm not exactly sure how I should continue.
Is there something I'm doing wrong, or is this an issue with the new version?

Thanks in advance!

**Comment by nblackburn** [comment]
If you could spin me up a example repo demonstrating this issue, that would be greatly helpful in getting to the bottom of this.

**Comment by AfraidKnot** [comment]
I am seeing the same thing.  It seem that babel is not being called after the template is converted as any ES2015 will cause the problem.

The App.vue file
```
<template><div></div></template>
<script>
    var fn = (a) => 'Hello, ' + a
</script>

```
Will generate
```
require.register("App.vue", function(exports, require, module) {
;(function(){
//

var fn = (a) => 'Hello, ' + a

})()

```
Using these dependencies
```
  "dependencies": {
    "browser-sync-brunch": "0.0.9",
    "vue": "^2.0.3"
  },
  "devDependencies": {
    "brunch": "^2.8.2",
    "vue-brunch": "^1.2.1",
    "babel-brunch": "^6.0.6",
    "babel-preset-es2015": "^6.16.0",
    "javascript-brunch": "^2.0.0",
    "sass-brunch": "^2.7.0"
  }
```

**Comment by nblackburn** [comment]
@AfraidKnot vue-brunch is basically a wrapper for vueify at the moment so it inherits any problems it may have. In this case babel-core is required as a peer dependency or it will not attempt to transpile anything within the `<script>` tag.

**Comment by AfraidKnot** [comment]
@nblackburn Thanks for your help, that was indeed the problem.

**Comment by nblackburn** [comment]
@AfraidKnot It's no bother, glad i was able to get it sorted for you.

**Comment by maximkott** [comment]
@nblackburn @AfraidKnot How did you solve this issue?

**Comment by nblackburn** [comment]
@maximkott You need to install babel-core as a peer dependency.

**Comment by maximkott** [comment]
Thank you for the quick response. Unfortunately installing `babel-core` does not fix the problem. `axios` http lib is not loading at all, but `lodash` does... I assume this package has additional issues. I've worked around that problem by creating a `lib/axios.js` file that imports **and** exports `axios`. Now I am able to load this lib from `.vue` files by import `lib/axios` instead of `axios`.

**Comment by nblackburn** [comment]
@maximkott There is 49 LoC so there isn't much that could go wrong with it. It is most likely an issue with Vueify that this is basically a wrapper for.

**Comment by matteocontrini** [comment]
I'm having the same issue. If a module is required for the first time in a .vue file, it seems that brunch isn't able to recognize that you need it, thus it doesn't build the module into the js bundle.

**Comment by PatKoperwas** [comment]
@DerJacques @matteocontrini @maximkott I ran into the same issue recently and I've found a solution. To get `imports` of npm modules to work try this:

Create a `.babelrc` file with 

```
{
  "presets": ["es2015", "es2016"]
}
```

I'm guessing `vueify` isn't respecting Brunch's Babel settings. But [vueify will happily use .babelrc](https://github.com/vuejs/vueify#es2015-with-babel)

Then if you want to get the object spread operator working (what lead me to this issue)

```
npm install --save-dev babel-plugin-transform-object-rest-spread

// .babelrc
{
  "presets": ["es2015", "es2016"],
  "plugins": ["transform-object-rest-spread"]
}
```

Then this should compile

```js
import { mapActions } from 'vuex'

export default {
  // ...
  methods: {
    ...mapActions([
      'increment' // map this.increment() to this.$store.dispatch('increment')
    ]),
    ...mapActions({
      add: 'increment' // map this.add() to this.$store.dispatch('increment')
    })
  }
}
```

**Comment by PatKoperwas** [comment]
**Edit:** Sigh module loading still doesn't quite work. ^ above will get spread working though

**Comment by aeons** [comment]
This is still a problem.

Why is this issue closed?

**Comment by elawn** [comment]
@aeons agreed.

**Comment by nblackburn** [comment]
@aeons It was closed because the original issue was resolved.

**Comment by aeons** [comment]
Hey,
I think only the es6 issues mentioned [the second comment](https://github.com/nblackburn/vue-brunch/issues/6#issuecomment-254623394) was fixed.

At least I cannot import stuff from npm directly in a `.vue` component, unless the import was already imported in a js file. I don't know if it is an issue with this library or vueify.

I have created a small repo reproducing my issue here https://github.com/aeons/vue-brunch-import

In the `.vue` file i `import 'axios';` but that fails unless axios is already referenced from initialize.js.
